UPGRADE.md

```diff
@@ -37,6 +37,158 @@ are terminated. In this way, Kong will serve new requests via the new
+git clone https://github.com/kong/kong
+cd kong
+git diff -w 2.0.0 2.1.0 kong/templates/nginx_kong*.lua
+```
+
+To produce a patch file, use the following command:
+
+```
+git diff 2.0.0 2.1.0 kong/templates/nginx_kong*.lua > kong_config_changes.diff
+```
+
+#### 3. Suggested Upgrade Path
+
+##### Upgrade from `0.x` to `2.1.0`
+
+The lowest version that Kong 2.1.0 supports migrating from is 1.0.0.
+If you are migrating from a version lower than 0.14.1, you need to
+migrate to 0.14.1 first. Then, once you are migrating from 0.14.1,
+please migrate to 1.5.0 first.
+
+The steps for upgrading from 0.14.1 to 1.5.0 are the same as upgrading
+from 0.14.1 to Kong 1.0. Please follow the steps described in the
+"Migration Steps from 0.14" in the [Suggested Upgrade Path for Kong
+1.0](#kong-1-0-upgrade-path), with the addition of the `kong
+migrations migrate-apis` command, which you can use to migrate legacy
+`apis` configurations.
+
+Once you migrated to 1.5.0, you can follow the instructions in the section
+below to migrate to 2.1.0.
+
+##### Upgrade from `1.0.0` - `2.0.0` to `2.1.0`
+
+**Postgres**
+
+Kong 2.1.0 supports a no-downtime migration model. This means that while the
+migration is ongoing, you will have two Kong clusters running, sharing the
+same database. (This is sometimes called the Blue/Green migration model.)
+
+The migrations are designed so that the new version of Kong is able to use
+the database as it is migrated while the old Kong cluster keeps working until
+it is time to decommission it. For this reason, the migration is split into
+two steps, performed via commands `kong migrations up` (which does
+only non-destructive operations) and `kong migrations finish` (which puts the
+database in the final expected state for Kong 2.1.0).
+
+1. Download 2.1.0, and configure it to point to the same datastore
+   as your old (1.0 to 2.0) cluster. Run `kong migrations up`.
+2. Once that finishes running, both the old (pre-2.1) and new (2.1.0)
+   clusters can now run simultaneously. Start provisioning 2.1.0 nodes,
+   but do not use their Admin API yet. If you need to perform Admin API
+   requests, these should be made to the old cluster's nodes. The reason
+   is to prevent the new cluster from generating data that is not understood
+   by the old cluster.
+3. Gradually divert traffic away from your old nodes, and into
+   your 2.1.0 cluster. Monitor your traffic to make sure everything
+   is going smoothly.
+4. When your traffic is fully migrated to the 2.1.0 cluster,
+   decommission your old nodes.
+5. From your 2.1.0 cluster, run: `kong migrations finish`.
+   From this point on, it will not be possible to start
+   nodes in the old cluster pointing to the same datastore anymore. Only run
+   this command when you are confident that your migration
+   was successful. From now on, you can safely make Admin API
+   requests to your 2.1.0 nodes.
+
+**Cassandra**
+
+Due to internal format changes, the Cassandra database for Kong 2.1.0 is
```

Comment:
It is unclear whether this section is describing a Kong upgrade only or a Cassandra version upgrade too. Might be helpful to clarify that.

Reply:
Fixed!